client/containers/Builder/Builder.js

```diff
@@ -55,14 +60,49 @@ class Builder extends Component {
     this.setState({ showModal: false })
   }
 
+  rotateButtonShow = (id, event) => {
+    const copy = [...this.state.pedalsOnBoard]
+    copy.forEach((elem, index) => {
+      if (elem.id === id) {
+        elem.showRotate = true
+      }
+    })
+
+    this.setState({ pedalsOnBoard: copy })
+  }
+
+  rotateButtonHide = (id, event) => {
+    const copy = [...this.state.pedalsOnBoard]
+
+    copy.forEach((elem, index) => {
+      if (elem.id === id) {
+        elem.showRotate = false
```

Comment:
You are mutating your object here. You need a combination of `Array.prototype.map` and `Object.assign` (or slice and spread).
https://redux.js.org/recipes/structuringreducers/immutableupdatepatterns#updating-an-item-in-an-array